docs/analytics-ga4/ga4-event-tracker.md

```diff
@@ -72,16 +63,14 @@ In some situations we want to track dynamic elements like the 'Show/hide all sec
 
 It is also complicated by the fact that the JavaScript that creates this element is imported from `govuk-frontend` and therefore can't be modified directly. In order to get attributes onto this link, we have to do the following.
 
-- pass attributes to the accordion for the 'Show/hide all' link using the `data_attributes_show_all` option
-  - this should contain an object with one or many keys, each with a value converted into JSON
-  - this is appended to the main component element as `data-show-all-attributes`
+- Enable tracking for an accordion using the `ga4_tracking` option
+  - In the ruby template, `data-show-all-attributes` is appended to the main component element with the GA4 JSON
 - the 'Show/hide all' link is created by govuk-frontend JavaScript
 - the accordion JavaScript reads `data-show-all-attributes`
-  - it creates a `data-` attribute on the 'Show/hide all' link for each key in the JSON
-  - in the example above, the result will be `data-ga4="{ "event_name": "select_content", "type": "accordion" }"`
-- wrap the accordion in the event tracking script
-  - tracking will be activated by clicks on elements with a `data-ga4` attribute
-  - it checks for an `aria-expanded` attribute if the clicked element or parent element contains a `data-ga4-expandable` value or the `gem-c-accordion` class. If it detects one of those, it sets the `action` of the GA data accordingly.
+  - it moves the `data-show-all-attributes` JSON to the 'Show/hide all' link, renaming the attribute to `data-ga4-event`
```

Comment:
Struggling a bit this afternoon post-lunch so I could be completely wrong about this, but checking in the component guide I can't see any attribute on the show/hide all link anymore. Has something broken or am I missing something obvious?

Reply:
Good spot - I am getting confused as well. It looks like the initial implementation of `show_all_attributes` takes a JSON containing JSONs, and this would apply each JSON as a `data-` attribute to the Show All link. However we just need to apply the one JSON directly on the data attribute now.

I've also just done a git blame and realised the `data-show-all-attributes` isn't needed for UA - you introduced it specifically for GA4. I got confused because the example wasn't in our GTM schema, and I think that's just because you wrote the code for this before we had a schema, so I thought the examples were UA stuff. (See [here](https://github.com/alphagov/govuk_publishing_components/commit/f193ef04feec77e79ca666d474e0221c6fd03ba9#diff-0353e1304e7acef271c38d2ac799bf23412749d788d92d9e3e8fab8b6b02a2bbR197))

Therefore I should be able to rename the attribute to `data-ga4-show-all-attributes`

Reply:
I'm getting confused about wether I can rename `data_show_all_attributes` in the accordions, but I think I figured it out.

Initially I thought `data_show_all_attributes`  was being used for UA, so I kept it in place. Then I saw that you introduced `data_show_all_attributes` specifically for GA4, so then I thought I could remove it. But it looks like another team have used this GA4 `data_show_all_attributes` attribute to add tracking to UA, so now I can't remove it.

@andysellick the code to add attributes to the Show All attributes link takes a JSON containing JSONs, so I'll have to update the template so I can either:
- Update the template so that it passes a JSON containing a JSON to `data-show-all-attributes` (this feels like adding technical debt)
- Duplicate the JavaScript used for`data-show-all-attributes` code, and make a specific one for GA4 with the attribute name `data-ga4-show-all-attributes` (I think this is preferable)

Reply:
@andysellick I think i've resolved this, see the latest commits...